### PR TITLE
add the map for dense tensor, test=develop

### PR DESCRIPTION
**Pull request by Shixiaowei02**
### PR types

Others
### PR changes

Others
### Describe

add the map for dense tensor, test=develop
Todo: replace the `infrt.dense_host_tensor_map` with the `phi.dense_tensor_map`

### CODE REVIEW

File: paddle/infrt/dialect/infrt/ir/infrt_dialect.cc [hunk 1]
```diff
@@ -91,7 +91,7 @@ mlir::Type InfrtDialect::parseType(::mlir::DialectAsmParser &parser) const {
         parser.getContext(), shape, elementType, lod_level);
   }
   if (keyword == "dense_tensor_map") {
-    return DenseTensorMapType::get(parser.getContext());
+    return DenseHostTensorMapType::get(parser.getContext());
```
Comment (DannyIsFunny) [inline paddle/infrt/dialect/infrt/ir/infrt_dialect.cc:94]: 这里是不是需要改回成  DenseTensorMapType 的解析

File: paddle/infrt/kernel/phi/dense_tensor_kernels.cc [hunk 2]
```diff
@@ -130,6 +145,53 @@ void PrintDenseTensor(::phi::DenseTensor* dense_tensor) {
   std::cout << "]\n";
 #undef PRINT_META_DATA
 }
+
+::infrt::phi::DenseTensorMap LoadParams(
+    host_context::Attribute<std::string> path) {
```
Comment (DannyIsFunny) [inline paddle/infrt/kernel/phi/dense_tensor_kernels.cc:150]: 这里能不能改成load param 从单个文件

File: paddle/infrt/kernel/phi/dense_tensor_kernels.cc [hunk 3]
```diff
@@ -130,6 +145,53 @@ void PrintDenseTensor(::phi::DenseTensor* dense_tensor) {
...
+  const auto& file_path = path.get();
+  std::cout << "loading params from: " << file_path << std::endl;
+  ::infrt::phi::DenseTensorMap map;
+
+  const std::string model_path = file_path + "/__model__";
+  auto pb_proto_prog = paddle::LoadProgram(model_path);
+  auto main_block = pb_proto_prog->blocks(0);
+
```
Comment (DannyIsFunny) [inline paddle/infrt/kernel/phi/dense_tensor_kernels.cc:158]:
```  C++
  std::string param_path = file_path + "/pdiparams";
   std::ifstream param_file(param_path, std::ios::binary);
```

File: paddle/infrt/kernel/phi/dense_tensor_kernels.cc [hunk 4]
```diff
@@ -130,6 +145,53 @@ void PrintDenseTensor(::phi::DenseTensor* dense_tensor) {
...
+  for (auto& var : main_block.vars()) {
+    if (var.name() == "feed" || var.name() == "fetch" || !var.persistable())
+      continue;
+    std::string param_path = file_path + "/" + var.name();
```
Comment (DannyIsFunny) [inline paddle/infrt/kernel/phi/dense_tensor_kernels.cc:162]: 删除这两句：
```C++
    std::string param_path = file_path + "/" + var.name();
    std::ifstream param_file(param_path, std::ios::binary);
```

File: paddle/infrt/kernel/phi/dense_tensor_kernels.cc (hunk 2 above)
Comment (Shixiaowei02) [reply paddle/infrt/kernel/phi/dense_tensor_kernels.cc:150]: done

File: paddle/infrt/kernel/phi/dense_tensor_kernels.cc (hunk 3 above)
Comment (Shixiaowei02) [reply paddle/infrt/kernel/phi/dense_tensor_kernels.cc:158]: ok

File: paddle/infrt/kernel/phi/dense_tensor_kernels.cc (hunk 4 above)
Comment (Shixiaowei02) [reply paddle/infrt/kernel/phi/dense_tensor_kernels.cc:162]: ok

File: paddle/infrt/dialect/infrt/ir/infrt_dialect.cc (hunk 1 above)
Comment (Shixiaowei02) [reply paddle/infrt/dialect/infrt/ir/infrt_dialect.cc:94]: the old dense tensor map needs to be removed

**Review by jiweibo — DISMISSED** [review]

**Review by jiweibo — APPROVED** [review]
LGTM

**Pull request merged**